### PR TITLE
[Cleanup] Pass reason for bulk message deletion to audit log

### LINKED ISSUE #5861: Specify audit log reason in cleanup cog
### What component of Red (cog, command, API) would you like to see improvements on?

Cleanup

### Describe the enhancement you're suggesting.

Delete messages endpoint supports specifying audit log reason and discord.py 2.0 added a `reason` kwarg to `Messageable.delete_messages(...)` in order to support it. I think that the cleanup cog should be specifying the audit log reason in all places where it deletes messages.

This change will require an addition of a `reason` kwarg to `redbot.core.utils.mod.mass_purge(...)` and usage of that new kwarg in all calls to this function in the Cleanup cog.

### Anything else?

What should the contents of the audit log reason be? Just whatever `get_audit_reason(...)` utility function returns?

### PR DESCRIPTION
**Pull request by leetfin**
### Description of the changes

Fixes #5861 

### Have the changes in this PR been tested?

No

### CODE REVIEW

File: redbot/core/utils/mod.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
 async def mass_purge(
     messages: List[discord.Message],
     channel: Union[discord.TextChannel, discord.VoiceChannel, discord.Thread],
+    reason: str = None,
```
Comment (Flame442) [inline redbot/core/utils/mod.py]: The linked issue specified this should be a kwarg, and I do think it would be better as one.

Comment (leetfin) [reply redbot/core/utils/mod.py]: Would `**reason: str,` be the correct way to specify this as a kwarg here?

Comment (Jackenmen) [reply redbot/core/utils/mod.py]: To make something a keyword-only argument, it needs to be specified after `*` symbol in the argument list, e.g:
```suggestion
    *,
    reason: Optional[str] = None,
```

File: redbot/core/utils/mod.py [hunk 2]
```diff
@@ -29,6 +31,8 @@ async def mass_purge(
         The messages to bulk delete.
     channel : `discord.TextChannel`, `discord.VoiceChannel`, or `discord.Thread`
         The channel to delete messages from.
+    reason : `str` - Optional
```
Comment (Jackenmen) [inline redbot/core/utils/mod.py]:
```suggestion
    reason : `str`, optional
```

**Review by Kowlin — APPROVED** [review]
Despite Discord not accepting reasons for mass deleted messages (while saying its supported by [their docs](https://discord.com/developers/docs/resources/channel#bulk-delete-messages)) this still gets send through properly. And is handled fine.

**Pull request merged**